Helm chart for kubernetes deployment

mikhailadvani opened the pull request:
Once merged, we can use github pages to publish the chart and update the documentation.

Closes #26

Strech commented:
@mikhailadvani Hey 👋 Thanks for your contrubision. I hope I will have enough time to make a review this weekend.

mikhailadvani commented:
@Strech I have added a few enhancements that I could think of related to security & reliability

mikhailadvani commented on helm-charts/sidekiq-prometheus-exporter/values.yaml:
```diff
@@ -0,0 +1,86 @@
+  type: "ConfigMap"
+
+envVars: {}
+  # REDIS_HOST: "localhost"
```
I have added the remaining default values in the values file in comments. the generic envVars approach allows additional/different environment variables as someone might want to use with a custom docker image. hence the all caps approach

Strech replied:
Yeah, I totally agree with caps and I think it's right, but my comment was about where to write docs, they should be placed on top of variable, not bottom. Bottom goes anything underneath to clarify what you can nest in the higher level key.

Here is an example https://github.com/bitnami/charts/blob/master/bitnami/redis/values.yaml

mikhailadvani replied:
it is supposed to be nested under envVars to allow dynamic usage through [this](https://github.com/mikhailadvani/sidekiq-prometheus-exporter/blob/392d9c0b896c534512dd2ad26792acb1ad373c60/helm-charts/sidekiq-prometheus-exporter/templates/deployment.yaml#L27-L32). I am not sure whether I understood what you meant in the redis example that you shared

Strech replied:
Ah, I got what you mean, yep, all good here

mikhailadvani commented on helm-charts/sidekiq-prometheus-exporter/values.yaml:
```diff
@@ -0,0 +1,86 @@
+  # annotations:
+  # Will add the provided map to the annotations for the created serviceAccount
+
+rbac:
```
updated based on https://helm.sh/docs/chart_best_practices/rbac/#using-rbac-resources

mikhailadvani commented on helm-charts/sidekiq-prometheus-exporter/values.yaml:
```diff
@@ -0,0 +1,86 @@
+  create: true
+  # The name of the ServiceAccount to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name:
```
updated based on https://helm.sh/docs/chart_best_practices/rbac/#using-rbac-resources

Strech commented:
@mikhailadvani I have some additional feedback. I think the best for the user will be an ability to spin up this helm with just a few values set, the ones which are crucial to the application.

I just tried to run the latest PR state in a console and get that:

```bash
$ helm install --generate-name --dry-run helm-charts/sidekiq-prometheus-exporter
Error: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "ServiceMonitor" in version "monitoring.coreos.com/v1"
```

mikhailadvani commented:
@Strech you get the error because you do not have the prometheus-operator installed. I have gone in for the prometheus-operator approach rather than the vanilla prometheus approach since it allows easier wiring in through service-discovery labels

Strech commented on helm-charts/sidekiq-prometheus-exporter/templates/service.yaml:
```diff
@@ -0,0 +1,16 @@
+metadata:
+  name: {{ include "sidekiq-prometheus-exporter.fullname" . }}
+  labels:
+{{ include "sidekiq-prometheus-exporter.labels" . | indent 4 }}
```
The issue is nesting, it's not visual and not obvious. I can give an example with `Deployment` which is 100% same with Service

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: {{ include "sidekiq-prometheus-exporter.fullname" . }}
  labels:
    {{- include "sidekiq-prometheus-exporter.labels" . | nindent 4 }}
```

mikhailadvani replied:
done

Strech commented:
> @Strech you get the error because you do not have the prometheus-operator installed. I have gone in for the prometheus-operator approach rather than the vanilla prometheus approach since it allows easier wiring in through service-discovery labels

Yes, you are right. I disable `serviceMonitor` and was able to dry-run it. But I mean in general, that I think that as less you need entities to spin the simple exporter as better. And all non-essential for application entities, in my opinion, should be possible to enable but disabled by default.

And again, if you turn them on – less configuration as possible to make them running.

mikhailadvani commented:
switched to serviceMonitor.enabled false and added an empty podAnnotations map for the ability to prometheus to scrape metrics in a non-operator setup

mikhailadvani commented:
addressed points 1 & 3. I do not agree on 2. keeping the service port as 80 with cluster IP keeps it simple for anyone consuming the service (rather than the pod) over http within the cluster as http://<service-name>. the port is an internal configuration of the setup and IMHO a candidate for abstraction. I look at the service as a load balancer/reverse proxy abstracting away the internal implementation and would prefer to stick to standard ports. your thoughts?

Strech commented:
@mikhailadvani your point sounds reasonable. Agree 👍

mikhailadvani commented:
all other comments have already been addressed

Strech commented:
@mikhailadvani Hey, I want to say thank you for your great contribution 👏 🎉

mikhailadvani commented:
you are welcome. more issues and PRs coming soon

Strech commented:
@mikhailadvani Currently I'm re-shaping cart, so be aware 👀
